src/contrib/mlir/compiler.cpp

```diff
@@ -300,15 +325,104 @@ void MLIRCompiler::lower_ng_dialect()
     m_engine = std::move(maybeEngine.get());
 }
 
+/// Creates target machine for current host.
+static llvm::Expected<std::unique_ptr<llvm::TargetMachine>> createDefaultTargetMachine()
+{
+    auto machineBuilder = llvm::orc::JITTargetMachineBuilder::detectHost();
+    if (!machineBuilder)
+    {
+        return machineBuilder.takeError();
+    }
+
+    // Retrieve host CPU sub-target features.
+    llvm::SubtargetFeatures subtargetFeatures;
+    llvm::StringMap<bool> featureMap;
+    llvm::sys::getHostCPUFeatures(featureMap);
+    for (auto& feature : featureMap)
+    {
+        subtargetFeatures.AddFeature(feature.first(), feature.second);
+    }
+
+    // Relocation model and code model are kept to default values.
+    machineBuilder->setCPU(llvm::sys::getHostCPUName());
+    machineBuilder->setCodeGenOptLevel(llvm::CodeGenOpt::Aggressive);
```

Comment:
Will this cause -O3 level ? We do set opt level for the ORCJit else where.

Reply:
It applies only to LLVM's codegen but I'm not sure what actually does. The optimization level of the LLVM optimizer is set up differently, using `populate...` MLIR APIs.

Reply:
Then rather keep this configurable based on `NGRAPH_MLIR_OPT_LEVEL`, otherwise debugging JIT'ed code will be hard.

Reply:
Probably better use the ocl flag for that. We could have our own -O{0,1,2,3}. I'll open a JR for that since we would have to also change LLVM optimizer pipeline for that.

Reply:
Not sure I follow. We already  have an env variable that controls the llvm pipeline. I just don't want confusion if I set OPT level to 0 and I still get optimized code while debugging. Can you verify that at least ? `NGRAPH_MLIR_OPT_LEVEL=0 NGRAPH_MLIR=1 NGRAPH_LLVM_OPTIONS="-print-after-all" ./unit-test --gtest-filter=CPU.mlir_*`